src/c++/perf_analyzer/triton_metrics_manager.cc

```diff
@@ -81,13 +85,22 @@ TritonMetricsManager::QueryTritonMetricsEveryNMilliseconds()
   }
 }
 
+void
+TritonMetricsManager::CheckQueryingStatus()
+{
+  if (query_loop_future_.wait_for(std::chrono::seconds(0)) ==
+      std::future_status::ready) {
+    query_loop_future_.get();
```

Comment:
The query loop shouldn't be ready when `CheckQueryingStatus()` is called. This means an exception was thrown.

`std::future::get()` propagates any potential exceptions thrown during the future's asynchronous execution.